### PR TITLE
Change to yarn

### PR DESCRIPTION
**Pull request by jzjiang**
- Changed npm usage to yarn
- Updated circle to run `yarn test`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -71,7 +72,7 @@
     "wdio-mocha-framework": "0.3.1",
     "webdriverio": "4.0.9",
     "whatwg-fetch": "0.10.1",
-    "yuidoc-lucid-theme": "git+ssh://git@github.com:Shopify/yuidoc-lucid-theme.git#c9c62ba",
+    "yuidoc-lucid-theme": "git+ssh://git@github.com/Shopify/yuidoc-lucid-theme.git#c9c62ba",
```
Comment (minasmart) [inline package.json]: Did this have to change? If it did, ok, but it's weird. NPM has waffled on this format of URI a lot :/

Comment (jzjiang) [inline package.json]: `yarn install` failed with the colon (https://github.com/yarnpkg/yarn/issues/513#issuecomment-254850558) :(

File: package.json [hunk 2]
```diff
@@ -19,8 +19,8 @@
     "test:livereload-server": "node scripts-graphql/livereload.js",
     "mocha": "mocha -u tdd $(yarn run mocha:reporter-args 2>&1 >/dev/null) .tmp/test/node-tests.js",
     "mocha:reporter-args": "test -n \"${CI}\" && >&2 echo --reporter xunit --reporter-options output=$CIRCLE_TEST_REPORTS/junit/mocha.xml || true",
-    "lint": "eslint --max-warnings 0 -c .eslintrc.json $(yarn run lint:reporter-args) src-graphql/ test-graphql/",
-    "lint:reporter-args": "[ -n \"${CI}\" ] && echo -o $CIRCLE_TEST_REPORTS/junit/eslint.xml -f junit",
+    "lint": "eslint --max-warnings 0 -c .eslintrc.json $(yarn run lint:reporter-args 2>&1 >/dev/null) src-graphql/ test-graphql/",
+    "lint:reporter-args": "test -n \"${CI}\" && >&s echo -o $CIRCLE_TEST_REPORTS/junit/eslint.xml -f junit",
```
Comment (minasmart) [inline package.json]: `&s` is not a file descriptor.

**Comment by jzjiang** [comment]
@minasmart does everything look okay to you?

**Pull request merged**